projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -234,6 +235,77 @@ async function checkForEscalation( issueReferences, commentBody, escalationNote,
 	return true;
 }
 
+/**
+ * Add or update a label on the issue to indicate a number range of support references,
+ * once it has gathered more than 10 support references.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} repo - Repository name.
+ * @param {string} ownerLogin - Owner of the repository.
+ * @param {number} number - Issue number.
+ * @param {number} issueReferencesCount - Number of support references gathered in this issue.
+ * @returns {Promise<void>}
+ */
+async function addOrUpdateInteractionCountLabel(
+	octokit,
+	repo,
+	ownerLogin,
+	number,
+	issueReferencesCount
+) {
+	const ranges = [ 50, 20, 10 ];
+
+	// Check if our issue has issues in one of the ranges where we want to label it.
+	const issueRange = ranges.find( range => issueReferencesCount > range );
+
+	// Bail if our issue hasn't gathered enough support references to warrant a label.
+	if ( ! issueRange ) {
+		return;
+	}
+
+	// Name of the label we want to add to this issue.
+	const interactionCountLabel = `[Interaction #] > ${ issueRange }`;
+
+	debug(
+		`gather-support-references: Issue #${ number } has gathered ${ issueReferencesCount } support references. It deserves a label, "${ interactionCountLabel }"`
+	);
+
+	// Check if the issue already has this label.
+	const labels = await getLabels( octokit, ownerLogin, repo, number );
+	const existingInteractionCountLabel = labels.find( label =>
```

Comment:
Should we use `.filter()` here instead of `.find()` in case someone manually messes with the labels and winds up with more than one on the issue?

That'd need some changes below too to handle the value being an array.

Reply:
That's a good point. It complicates the logic a bit, but here we go:
0de6d402d31216a62dd233df1eaab48f76e41f9b